src/pilot/assign_role.py

```diff
@@ -381,18 +385,39 @@ def get_raid_controller_physical_disk_ids(drac_client, raid_controller_fqdd):
         key=physical_disk_id_to_key)
 
 
-def define_storage_logical_disks(drac_client, raid_controller_name):
+def define_storage_logical_disks(drac_client, raid_controller_ids):
+    boss_controller = None
+    raid_controller = None
     all_physical_disks = drac_client.list_physical_disks()
 
+    if len(raid_controller_ids) > 1:
+        for cntrl in raid_controller_ids:
+            if drac_client.is_boss_controller(cntrl):
+                boss_controller = cntrl
+            else:
+                raid_controller = cntrl
+    else:
+        if drac_client.is_boss_controller(raid_controller_ids[0]):
+            boss_controller = raid_controller_ids[0]
+        else:
+            raid_controller = raid_controller_ids[0]
+
     # Get the drives controlled by the RAID controller
     raid_cntlr_physical_disks = []
+    boss_cntrl_disks = []
     for disk in all_physical_disks:
-        if disk.controller == raid_controller_name:
+        if disk.controller == boss_controller:
+            boss_cntrl_disks.append(disk)
+        elif disk.controller == raid_controller:
             raid_cntlr_physical_disks.append(disk)
 
     # Make sure we have enough drives attached to the RAID controller to create
     # a RAID1
-    num_raid_cntlr_physical_disks = len(raid_cntlr_physical_disks)
+
+    if boss_cntrl_disks:
+        num_raid_cntlr_physical_disks = len(boss_cntrl_disks)
+    else:
+        num_raid_cntlr_physical_disks = len(raid_cntlr_physical_disks)
```

Comment:
as long as one raid controller has at least 2 disks, then we're fine.  if no raid controller has at least 2 disks, then emit the error below

Reply:
Agreed. Done the required changes.